device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientOptions.java

```diff
@@ -31,4 +37,12 @@
     @Setter
     @Getter
     public SSLContext sslContext;
+
+    /**
+     * The convention to be used for convention based operations.
+     */
+    @Setter
+    @Getter
+    @Builder.Default
```

Comment:
This is interesting, so now any time ClientOptions is created using a builder, the payload convention is set to our default convention (unless explicitly set to something else). Is that something we want to do? Should convention be tied to the existence of a model Id instead?

Reply:
That is a good question. In all cases only convention aware methods would use this so having it set to default makes sure if someone use a convention aware method without setting the model id it would always use our convention.

Do we do this in .NET?